File: ios/PaymentUtils.swift
import Foundation
import KhaltiCheckout

/// Utilities for payment operations
class PaymentUtils {

    /**
     * Validates payment arguments
     */
    static func validatePaymentArgs(_ args: PaymentArgs) throws {
        if args.publicKey.isEmpty {
            AppLogger.e("PaymentValidation", "Public key is required")
            throw InvalidConfigException()
        }

        if args.pidx.isEmpty {
            AppLogger.e("PaymentValidation", "PIDX is required")
            throw InvalidConfigException()
        }

        if !["TEST", "PROD"].contains(args.environment.uppercased()) {
            AppLogger.e("PaymentValidation", "Invalid environment: \(args.environment)")
            throw InvalidConfigException()
        }
    }

    /**
     * Parses environment string to Khalti environment
     */
    static func parseEnvironment(_ environment: String) -> Environment {
        switch environment.uppercased() {
        case "PROD":
            return .PROD
        default:
            return .TEST
        }
    }

    /**
     * Creates payment configuration from arguments
     */
    static func createPaymentConfig(_ args: PaymentArgs, _ environment: Environment)
        -> KhaltiPayConfig
    {
        return KhaltiPayConfig(
            publicKey: args.publicKey,
            pIdx: args.pidx,
            environment: environment
        )
    }
}

File: android/src/main/java/expo/modules/khalticheckout/utils/PaymentUtils.kt
package expo.modules.khalticheckout.utils

import com.khalti.checkout.data.Environment
import com.khalti.checkout.data.KhaltiPayConfig
import expo.modules.khalticheckout.AppLogger
import expo.modules.khalticheckout.constants.KhaltiConstants
import expo.modules.khalticheckout.exceptions.InvalidConfigException
import expo.modules.khalticheckout.models.PaymentArgs

/**
 * Utility functions for payment configuration and validation.
 */
object PaymentUtils {

  /**
   * Validates payment arguments.
   *
   * @throws InvalidConfigException if arguments are invalid.
   */
  fun validatePaymentArgs(args: PaymentArgs) {
    if (args.publicKey.isBlank()) {
      throw InvalidConfigException("Public key is required")
    }
    if (args.pidx.isBlank()) {
      throw InvalidConfigException("PIDX is required")
    }
    if (args.publicKey.length < 10) {
      throw InvalidConfigException("Invalid public key format")
    }
  }

  /**
   * Parses environment string to Khalti Environment enum.
   */
  fun parseEnvironment(environment: String): Environment {
    return when (environment.uppercase()) {
      KhaltiConstants.ENV_PROD -> Environment.PROD
      KhaltiConstants.ENV_TEST, "" -> Environment.TEST
      else -> {
        AppLogger.w("PaymentUtils", "Unknown environment '$environment', defaulting to TEST")
        Environment.TEST
      }
    }
  }

  /**
   * Create payment configuration.
   */
  fun createPaymentConfig(args: PaymentArgs, environment: Environment): KhaltiPayConfig {
    return KhaltiPayConfig(
      publicKey = args.publicKey,
      pidx = args.pidx,
      environment = environment
    )
  }
}
